Gems/ExpressionEvaluation/Code/Source/ElementInformationSerializer.inl

```diff
@@ -0,0 +1,181 @@
+                    {
+                        return context.Report(result, "ElementInformationSerializer::Load failed to load a value matched the "
+                            "reported AZ TypeId. The C++ declaration may have been deleted or changed.");
+                    }
+
+                    result.Combine
+                        ( ContinueLoadingFromJsonObjectField(AZStd::any_cast<void>(&storage), typeId, inputValue, "Value", context));
+                    outputDatum->m_extraStore = storage;
+                }
+            }
+            // any storage end
+
+            return context.Report(result, result.GetProcessing() != JSR::Processing::Halted
+                ? "ElementInformationSerializer Load finished loading ElementInformation"
+                : "ElementInformationSerializer Load failed to load ElementInformation");
+        }
+
+        JsonSerializationResult::Result Store
+            ( rapidjson::Value& outputValue
+            , const void* inputValue
+            , const void* defaultValue
+            , const Uuid& valueTypeId, JsonSerializerContext& context) override
+        {
+            namespace JSR = JsonSerializationResult;
+
+            AZ_Assert(valueTypeId == azrtti_typeid<ElementInformation>(), "ElementInformation Store against value typeID that "
+                "was not ElementInformation");
+            AZ_Assert(inputValue, "ElementInformation Store against null inputValue pointer ");
+
+            auto inputScriptDataPtr = reinterpret_cast<const ElementInformation*>(inputValue);
+            auto defaultScriptDataPtr = reinterpret_cast<const ElementInformation*>(defaultValue);
+
+             if (defaultScriptDataPtr)
+             {
+                 if (inputScriptDataPtr->m_id == defaultScriptDataPtr->m_id
+                 && AZ::Helpers::CompareAnyValue(inputScriptDataPtr->m_extraStore, defaultScriptDataPtr->m_extraStore))
+                 {
+                     return context.Report
+                             ( JSR::Tasks::WriteValue, JSR::Outcomes::DefaultsUsed, "ElementInformation Store used defaults for "
+                                "ElementInformation");
+                 }
+             }
+
+            JSR::ResultCode result(JSR::Tasks::WriteValue);
+            outputValue.SetObject();
+
+            result.Combine(ContinueStoringToJsonObjectField
+                ( outputValue
+                , "Id"
+                , &inputScriptDataPtr->m_id
+                , defaultScriptDataPtr ? &defaultScriptDataPtr->m_id : nullptr
+                , azrtti_typeid<decltype(inputScriptDataPtr->m_id)>()
+                , context));
+
+            if (!inputScriptDataPtr->m_extraStore.empty())
+            {
+                rapidjson::Value typeValue;
+                result.Combine(StoreTypeId(typeValue, inputScriptDataPtr->m_extraStore.type(), context));
+                outputValue.AddMember
+                    ( rapidjson::StringRef(JsonSerialization::TypeIdFieldIdentifier)
```

Comment:
Nitpick, repeated: for consistency the type id field should be the first value in an object. Can the previous "Id" field be moved until after this block that writes the type id field?

Reply:
I do have the type id be the first value of an object, but this isn't a typeid yet, the type id is written first when I hit the next object which contains it.